content/en/blog/2024/prom-and-otel/index.md

```diff
@@ -0,0 +1,572 @@
+
+Luckily, the OTel Operator’s Target Allocator (TA) is able to help with 
+some of this. For instance, it can automatically drop any targets it knows won’t 
+be scraped. Whereas if you shard with `hashmod`, you'll need to [update your
+config based on the number of replicas you have](https://www.robustperception.io/scaling-and-federating-prometheus/).
+Plus, if you’re already collecting Prometheus metrics about your 
+Kubernetes infrastructure, using the TA is a great option.
+
+The Target Allocator is part of the OTel Operator. The OTel Operator is a [Kubernetes Operator](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/) that:
+* Manages the [OpenTelemetry Collector](/docs/collector/)
+* Injects and configures [auto-instrumentation](https://www.honeycomb.io/blog/what-is-auto-instrumentation) into your pods
+
+In fact, the Operator creates two new [custom resource](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) (CR) types in Kubernetes to 
+support this functionality: the [OpenTelemetry Collector CR](https://github.com/open-telemetry/opentelemetry-operator#getting-started), and the [Auto-Instrumentation CR](https://github.com/open-telemetry/opentelemetry-operator#opentelemetry-auto-instrumentation-injection).
+
+Today, we will be focusing on the Target Allocator. The TA is an optional 
+component of the Operator’s OTel Collector management capabilities.
+
+In a nutshell, the Target Allocator is a mechanism for decoupling the service 
+discovery and metric collection functions of Prometheus in a way that allows 
+them to be scaled independently. The OTel Collector manages Prometheus Metrics 
+without needing to install Prometheus. The TA manages the configuration of the 
+Collector’s Prometheus Receiver.
+
+The Target Allocator serves two functions:
+* Even distribution of Prometheus targets among a pool of OTel Collectors
+* Discovery of Prometheus custom resources
+
+Let’s dig into each of these.
+
+#### Even distribution of Prometheus targets
+The Target Allocator’s first job is to discover targets to scrape and OTel 
+Collectors to allocate targets to. It does so as follows:
+
+1. The Target Allocator finds all of the the metrics targets to scrape
+2. The Target Allocator finds all of the available Collectors
+3. The Target Allocator determines which Collectors scrape which metrics
+4. The Collectors query the Target Allocator to find out what metrics to scrape
+5. The Collectors scrape their assigned targets
+
+This means that the OTel collectors -- not a Prometheus scraper -- collect 
+the metrics.
+
+A **Target** is an endpoint that supplies Metrics for Prometheus to store. A 
+**Scrape** is the action of collecting Metrics through an HTTP request from a 
+targeted instance, parsing the response, and ingesting the collected samples 
+to storage.
+
+![Data flow from Target Allocator to the OTel Collector.](target-allocator-sharding.png)
```

Comment:
```suggestion
```mermaid
flowchart RL
  pm(PodMonitor)
  sm(ServiceMonitor)
  ta(Target Allocator)
  oc1(OTel Collector)
  oc2(OTel Collector)
  oc3(OTel Collector)
  ta --> pm
  ta --> sm
  oc1 --> ta
  oc2 --> ta
  oc3 --> ta
  sm ~~~|"1. Discover Prometheus Operator CRs"| sm
  ta ~~~|"2. Add job to TA scrape configuration"| ta
  oc3 ~~~|"3. Add job to OTel Collector scrape configuration"| oc3
````
```